main.py

```diff
@@ -135,6 +136,51 @@ async def handle_augments(screenshot: ndarray, adb_instance: ADB):
     await asyncio.sleep(1)
 
 
+async def check_surrender_state(adb_instance: ADB, screenshot: ndarray, config: AluneConfig):
+    """
+    Check if we're able to surrend from the current game state
+
+    Args:
+        adb_instance: The adb instance to process the surrend phase.
+        screenshot: The current screen.
+        config: An instance of the alune config to use.
+    """
+    # If the surrender feature is enabled, we expand the top bar to check if
+    # we're in a surrend-possible phase
+    if config.get_auto_surrender():
+        await adb_instance.click_button(Button.expand_top_bar)
+        await asyncio.sleep(1)
+        is_phase_3_2 = screen.get_on_screen(screenshot, Image.PHASE_3_2_FULL)
+        if is_phase_3_2:
+            random_delay = config.get_auto_surrender_random_delay()
+            if random_delay == 0:
+                logger.info("We're in a phase we can surrend ! No delay have been set in config file.")
+                logger.info("Surrendering now !")
+                surrender_game(adb_instance)
+            else:
+                logger.info("We're in a phase we can surrend. A random delay have been set in config file.")
+                logger.info(f"Surrendering in {random_delay} second(s) !")
+                surrender_timer = Timer(random_delay, await surrender_game(adb_instance))
+                surrender_timer.start()
+            return True
+    return False
+
+
+async def surrender_game(adb_instance: ADB):
+    """
+    Surrends the current game
+
+    Args:
+        adb_instance: The adb instance to process the surrend phase.
+    """
+    await adb_instance.send_key(111)  # Send escape key, this open the settings menu
+    await asyncio.sleep(2)
+    await adb_instance.click_button(Button.surrender)  # Click the surrend button, this open a choice window
+    await asyncio.sleep(2)
+    await adb_instance.click_button(Button.check_surrender)  # Confirm surrend
+    await asyncio.sleep(5)
```

Comment:
Please remove the comments here. The code is self-documenting - "click button ( surrender )".